src/kvstore/raftex/SnapshotManager.cpp

```diff
@@ -0,0 +1,116 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+#include "kvstore/raftex/SnapshotManager.h"
+#include "base/NebulaKeyUtils.h"
+#include "kvstore/raftex/RaftPart.h"
+
+DEFINE_int32(snapshot_worker_threads, 4, "Threads number for snapshot");
+DEFINE_int32(snapshot_io_threads, 4, "Threads number for snapshot");
+DEFINE_int32(snapshot_send_retry_times, 3, "Retry times if send failed");
+DEFINE_int32(snapshot_send_timeout_ms, 60000, "Rpc timeout for sending snapshot");
+
+namespace nebula {
+namespace raftex {
+
+SnapshotManager::SnapshotManager() {
+    executor_.reset(new folly::IOThreadPoolExecutor(FLAGS_snapshot_worker_threads));
+    ioThreadPool_.reset(new folly::IOThreadPoolExecutor(FLAGS_snapshot_io_threads));
```

Comment:
why use private thread pool? 
maybe executor is threadmanager is better, because it is not process the network io.

Reply:
Because the threads in executor will be blocked during sending snapshot.

Reply:
maybe can define a thread manager here which owned by SnapshotManager

Reply:
ThreadManager is not a good thread pool implementation.  I think IOThreadPool is better, because it holds a event loop inside.